projects/plugins/jetpack/extensions/blocks/donations/donations-view/edit.js

```diff
@@ -0,0 +1,109 @@
+/**
+ * Internal dependencies
+ */
+import DonationsContext from '../common/context';
+import { ANNUAL_DONATION, MONTHLY_DONATION, ONE_TIME_DONATION } from '../common/constants';
+
+/**
+ * External dependencies
+ */
+import classNames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { useContext } from '@wordpress/element';
+import { __, sprintf } from '@wordpress/i18n';
+import { InnerBlocks } from '@wordpress/block-editor';
+
+const types = {
+	[ ONE_TIME_DONATION ]: {
+		heading: __( 'Make a one-time donation', 'jetpack' ),
+		cta: __( 'Donate', 'jetpack' ),
+	},
+	[ MONTHLY_DONATION ]: {
+		heading: __( 'Make a monthly donation', 'jetpack' ),
+		cta: __( 'Donate monthly', 'jetpack' ),
+	},
+	[ ANNUAL_DONATION ]: {
+		heading: __( 'Make a yearly donation', 'jetpack' ),
+		cta: __( 'Donate yearly', 'jetpack' ),
+	},
+};
+
+const Edit = props => {
+	const { attributes, setAttributes } = props;
+	const { type } = attributes;
+	const { heading, cta } = types[ type ];
+	const { activeTab, fallbackLinkUrl, products } = useContext( DonationsContext );
+	const planId = products[ type ];
+
+	setAttributes( { fallbackLinkUrl, planId } );
```

Comment:
I wonder if this might cause a render loop.
We might want to insert it into a `useEffect`, with `planId` as dependency so that it would only execute when `planId` changes.

Also as noted elsewhere, do we need `fallbackLinkUrl` in this child block?

Reply:
It doesn't cause a render loop because we use the value from the context, which isn't tied to the block's attributes. We read from the context and store in the block attributes with every render. A render is triggered by the useContext hook when the context changes AFAIK. 
Wouldn't It be redundant to add a `useEffect` for this concrete case?

Reply:
> Also as noted elsewhere, do we need fallbackLinkUrl in this child block?

I'll have a deeper look into this, we might not need the fallbackLinkUrl anywhere. From what I've seen in the previous version of the block, the fallbackLinkUrl is used when: `! jetpack_is_frontend()` so that the render function returns buttons with a link to fallbackLinkUrl. Because otherwise, the fallbackLinkUrl is always either overwritten with the real URL or in the case in which we don't have a valid plan id, we aren't saving anything.